File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
    var answer = true;
    const map = new Map();
    phone_book.forEach((number, index) => {
        map.set(number, 1);
    })
    
    for(const number of phone_book) {
        for(let i=1;i<number.length;i++){
            if(map.has(number.slice(0,i))) {
                return false;
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    unordered_map<string, int> map;
    for(string s : phone_book) {
        map.insert(make_pair(s, 1));
    }

    for(string s : phone_book) {
        for(int i=1;i<s.length();i++) {
            if(map.find(s.substr(0,i)) != map.end()) {
                answer = false;
            }
              
        }
    }
    return answer;
}